staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -298,7 +299,12 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		}
 
 		if err := r.watchHandler(w, &resourceVersion, resyncerrc, stopCh); err != nil {
-			if err != errorStopRequested {
+			switch err {
+			case errorStopRequested:
+				// no extra log message
+			case ErrStaleWatch:
```

Comment:
this compares the human readable message (next to anything else of the error). If you look what the error actually is:
```
	return &StatusError{metav1.Status{
		Status:  metav1.StatusFailure,
		Code:    http.StatusGone,
		Reason:  metav1.StatusReasonExpired,
		Message: message,
	}}
```

we should probably just compare the reason (meant to be machine readable). There is even a func `IsResourceExpired` for that. Then you also don't have to export the `ErrStaleWatch` from here, and we avoid coupling of our client library with some low level storage library for etcd.

Reply:
thanks! I did not look that far, now I know :)